src/globalstate.h

```diff
@@ -18,9 +18,30 @@
 #include <src/gasman.h>
 
 #define MAXPRINTDEPTH 1024L
+#define TLS_NUM_EXTRA 256
 
-typedef struct GlobalState
+typedef struct GAPState
 {
+#if defined(HPCGAP)
+  int threadID;
+  void *threadLock;
+  void *threadSignal;
+  void *acquiredMonitor;
+  unsigned multiplexRandomSeed;
+  void *currentRegion;
+  void *threadRegion;
+  void *traversalState;
+  Obj threadObject;
+  Obj tlRecords;
+  Obj lockStack;
+  int lockStackPointer;
+  Obj copiedObjs;
+  Obj interruptHandlers;
+  void *CurrentHashLock;
+  char *CurrFuncName;
+  int DisableGuards;
+#endif
```

Comment:
So now `GAPState` contains purely HPC-GAP specific stuck. Luckily, this at last does not require any custom types. But I still wonder if we should distinguish between `GAPState` (which is used by GAP and HPC-GAP) and `ThreadLocalStorage` (which only HPC-GAP would know about), and the latter would be a struct that contains a `GAPState` member, plus the stuff above, plus the extra vars at the bottom.